src/borg/cache.py

```diff
@@ -70,6 +70,10 @@ def __init__(self, repository, key, manifest, path=None, sync=True, do_files=Fal
         self.key = key
         self.manifest = manifest
         self.path = path or os.path.join(get_cache_dir(), repository.id_str)
+        if os.environ.get('BORG_UNIQ_HOSTNAME'):
```

Comment:
another thing that came to my mind:
if we create and store a uuid clientside, we could use that and not need the user to assert uniqueness of hostname.